### DISCUSSION TITLE
Now have Lilygo-tdisplay-s3 working with digital pdm mic

### DISCUSSION THREAD

**Discussion by prschguy1** [PlummersSoftwareLLC/NightDriverStrip#567] [Ideas]
Had been looking at the build env for Lilygo-tdisplay-s3 and it looked like it could use some help. Now have the display working with digital PDM mic. Have added support for 16mb and psram. Have 44 key remote working with it. Board is working well with very stable wifi functions. Might be an alternative for those having trouble with M5 stick? Board is not without it's short comings though. Being a s3 board it seems to only be able to handle 2 led channels. Additionally, as Spectrum uses DMA, there is a conflict with this boards Parallel display. I am not sure if there is a work around for that or not. My reading says not. So at this point, only useful for strip and fan builds. Still early in testing but looks promising.. Board flashes easily so it's a good starting point for us non programmers wanting to work with S3 envs. Glad to have it in my collection all the same.

https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/69419979/9d143588-ec6c-4375-a902-ba29d91ab7e4

**Comment by davepl** [comment]
Nice!  I use the feather s3 for my all-year-Christmas lights.  So the ledstrip_feather config is usually pretty up to date and well tested!

**Reply by prschguy1** [reply]
Thanks Dave, that means a lot coming from you. Had flashed feather a few times before, but never used it before. Reason being as I was looking for stand alone builds, and every time I enabled audio on it, it blew up. Haven't gotten into streaming just yet. I am learning though, so might not be far off. I now know how to pdm mic to work, so may look at it again. Think I have esp32-s3-devkitc-1-n16r8v pretty well figured out with display,remote,  and pdm mic, so will probably look at that first. Parallel display was a short coming on lily go t display, so now will try with a simple spi display. Who knows, might even work with my hub75 display?

**Comment by robertlipe** [comment]
>  esp32-s3-devkitc-1-n16r8v

I use those bare modules a lot now. I have changes pending to add those as
a 'raw' device where we can support the raw module as it might appear on
multiple dev boards. I've had great luck with them. There's a lot of
S3-specific voodoo I'd like to add to the code in time.

~/.platformio/penv/bin/pio run --target upload -e yd-esp32-s3-demo

Is my 'goto' test target for doing random low-level work in the project. I
suspect that replacing the bottom layer code that actually clocks the
WS281x and drives the HUB75 with S3 code that really knows how to exercise
the RMT from DMA and use the LCD device to handle the 3/4/5/9-bit parallel
stuff on HUB75 would really help these parts even more, but I don't NEED 16
pins of 2812's via the dedicated GPIO
<https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/dedic_gpio.html>
controller that's in the newer chips[1].  Both FastLED and SmartMatrix seem
to have kind of lost their way on new hardware support and are stuck in
time tarpits trying to get 4,000 LEDS on an ATMEga working or something
crazy.

I like my 44 key remotes, too, but have held off submitting the changes to
just fix the silly typos that are in that file to fix it.



[1] That MIGHT be the way to get the single-core ESP32-C3 flying on this
project... Another project I'd love to explore, but I'm more interested in
scaling UP than DOWN. When the n16r8 boards are < $5, my interest in going
*down* is pretty low. There's a limit to my frugality.
